app/controllers/admin/daily_reports_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::DailyReportsController < Admin::BaseController
+  # GET /daily_reports
+  def index
+    all_reports = DailyReport.recent.includes(:user, :course)
+                             .by_course_filter(params[:course_id])
+                             .on_day(params[:filter_date])
+                             .by_user(params[:user_id])
+
+    @pagy, @daily_reports = pagy(all_reports)
+  end
+
+  # GET /daily_reports/:id
+  def show
+    @daily_report = DailyReport.find_by(id: params[:id],
+                                        status: Settings.daily_report
+                                        .status.submitted)
+    return if @daily_report
+
+    flash[:danger] = t(".report_not_found")
+    redirect_to admin_daily_reports_path
+  end
```

Comment:
```suggestion
  def show; end
```
Logic load_daily_report này e tách ra before_action nha